Sources/Workspace/Workspace.swift

```diff
@@ -516,6 +516,18 @@ public class Workspace {
             RepositoryPackageConstraint(
                 container: dependency.repository, requirement: requirement))
 
+        // Add unversioned constraint for edited packages.
+        let currentManifests = loadDependencyManifests(rootManifests: rootManifests, engine: engine)
+        for (externalManifest, managedDependency) in currentManifests.dependencies {
+            switch managedDependency.state {
+            case .checkout: continue
+            case .unmanaged, .edited: break
+            }
+            let specifier = RepositorySpecifier(url: externalManifest.url)
+            let dependencies = externalManifest.package.dependencyConstraints()
+            constraints += [RepositoryPackageConstraint(container: specifier, requirement: .unversioned(dependencies))]
```

Comment:
I'm still reading the various side-bits of code and the related tests. It looks like in the tests, pinning and editing are dealt with orthogonally. When I was manually poking at a local test setup, I noticed that Package.pins listed the packages that I'd "edit"ed. Still reading to understand where/how Pins work and all the little tendrils there, but it surprised me - for some reason I had it in my head that an edit'd package wouldn't show up in pins.